src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformRepositoriesHelper.kt

```diff
@@ -142,33 +143,91 @@ class IntelliJPlatformRepositoriesHelper(
         return repository
     }
 
-    internal fun createLocalIvyRepository(repositoryName: String, action: IvyRepositoryAction = {}) = repositories.ivy {
-        name = repositoryName
+    internal fun createLocalIvyRepository(repositoryName: String, action: IvyRepositoryAction = {}): IvyArtifactRepository {
+        val ivyLocationPath = providers.localPlatformArtifactsPath(rootProjectDirectory)
+
+        return createExclusiveIvyRepository(
+            repositoryName,
+            setOf(
+                Dependencies.LOCAL_IDE_GROUP,
+                Dependencies.LOCAL_PLUGIN_GROUP,
+                Dependencies.LOCAL_JETBRAINS_RUNTIME_GROUP
+            ),
+            repositories,
+            ivyLocationPath,
+            // We can not know it, since this repo is used to store all kinds of things located in unrelated paths.
+            artifactLocationPath = null,
+            action
+        )
+    }
 
-        // Location of Ivy files generated for the current project.
-        val localPlatformArtifactsPath = providers.localPlatformArtifactsPath(rootProjectDirectory)
-        ivyPattern("${localPlatformArtifactsPath.pathString}/[organization]-[module]-[revision].[ext]")
+    companion object {
+        internal fun createExclusiveIvyRepository(
+            repositoryName: String,
+            exclusiveGroups: Set<String> = emptySet(),
+            repositories: RepositoryHandler,
+            ivyLocationPath: Path,
+            artifactLocationPath: Path?,
+            action: IvyRepositoryAction = {},
+        ): IvyArtifactRepository {
+            val repository = createIvyArtifactRepository(
+                repositoryName, repositories, ivyLocationPath, artifactLocationPath
+            )
+
+            // For performance reasons make it exclusive.
+            // https://docs.gradle.org/current/userguide/declaring_repositories_adv.html#declaring_content_exclusively_found_in_one_repository
+            repositories.exclusiveContent {
```

Comment:
So this is also an improvement over #1783 I noticed that there is a dedicated API for telling Gradle "look for this group only here" which is probably the safest option possible, especially if combined with prefixed groups with "com.jetbrains".

Reply:
Btw, if we pre-create that new dynamic Ivy repository here as well, we can also make it exclusive, it may save a couple of seconds on the build. But again, I am not sure that creating a repo with broken paths is something we should be doing.